diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,183 @@
+from selenium.webdriver.support.ui import WebDriverWait
+from xblock.fields import String
+from xblockutils.base_test import SeleniumXBlockTest
+from xblockutils.resources import ResourceLoader
+import time
+# Studio adds a url_name property to each XBlock but Workbench doesn't.
+# Since we rely on it, we need to mock url_name support so it can be set via XML and
+# accessed like a normal field.
+from diagnostic_feedback.quiz import QuizBlock
+QuizBlock.url_name = String()
+
+loader = ResourceLoader(__name__)
+
+
+class DiagnosticFeedbackBaseTest(SeleniumXBlockTest):
+    """
+    The new base class for integration tests.
+    Scenarios can be loaded and edited on the fly.
+    """
+    default_css_selector = 'div.diagnostic-feedback'
+
+    def _get_previous_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[0]
+
+    def _get_previous_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#previous']")
+
+    def _get_next_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[1]
+
+    def _get_next_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#next']")
+
+    def wait_until_disabled(self, elem):
+        """ Wait until the given element is disabled """
+
+        wait = WebDriverWait(elem, self.timeout)
+        wait.until(lambda e: elem.get_attribute('class') == 'disabled',
+                   u"{} should be disabled".format(elem.text))
+
+    def wait_until_enabled(self, elem):
```

Comment:
@jareerahsan I think you could use [`wait_until_clickable`](https://github.com/edx/xblock-utils/blob/b4f9b51146c7fafa12f41d54af752b8f1516dffd/xblockutils/base_test.py#L67) from `SeleniumXBlockTest` (instead of defining a new method here) to check if an element is enabled. It does one additional thing, which is to check if the `elem` in question is visible to the user; but that shouldn't be a problem (unless you need to check if an element is enabled and *not* visible to the user).

If there's a specific use case that you need this method for, it might make sense to try and modify the code so that the check performed by the `lambda` function can change from `elem.get_attribute('class') == ''` to `e.is_enabled()`. That would make the test (and the code being tested) more robust. For example, the existing check would fail if the `elem` in question had other classes than `disabled` associated with it.